src/System.Security.Cryptography.Encoding/src/System/Security/Cryptography/Base64Transforms.cs

```diff
@@ -96,143 +103,211 @@ public class FromBase64Transform : ICryptoTransform
 
-            if (effectiveCount + _inputIndex < 4)
+            // To less data to decode
+            if (bytesToTransform < InputBlockSize)
             {
+                // reinitialize the transform
                 Reset();
+
                 return Array.Empty<byte>();
             }
 
-            byte[] result = ConvertFromBase64(temp, effectiveCount);
+            int outputSize = GetOutputSize(bytesToTransform, tmpBuffer);
+            byte[] output = new byte[outputSize];
+
+            ConvertFromBase64(tmpBuffer, output, out int consumed, out int written);
+            Debug.Assert(consumed == bytesToTransform);
+            Debug.Assert(written == outputSize);
+
+            if (tmpBufferArray != null)
+            {
+                tmpBuffer.Clear();
+                ArrayPool<byte>.Shared.Return(tmpBufferArray);
+            }
 
             // reinitialize the transform
             Reset();
 
-            return result;
+            return output;
         }
 
-        private byte[] GetTempBuffer(byte[] inputBuffer, int inputOffset, int inputCount, out int effectiveCount)
+        private Span<byte> GetTempBuffer(Span<byte> inputBuffer, Span<byte> tmpBuffer)
         {
-            byte[] temp;
-
-            if (_whitespaces == FromBase64TransformMode.IgnoreWhiteSpaces)
+            if (_whitespaces == FromBase64TransformMode.DoNotIgnoreWhiteSpaces)
             {
-                temp = DiscardWhiteSpaces(inputBuffer, inputOffset, inputCount);
-                effectiveCount = temp.Length;
+                return inputBuffer;
             }
-            else
+
+            return DiscardWhiteSpaces(inputBuffer, tmpBuffer);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Span<byte> DiscardWhiteSpaces(Span<byte> inputBuffer, Span<byte> tmpBuffer)
+        {
+            int count = 0;
+
+            for (int i = 0; i < inputBuffer.Length; i++)
             {
-                temp = new byte[inputCount];
-                Buffer.BlockCopy(inputBuffer, inputOffset, temp, 0, inputCount);
-                effectiveCount = inputCount;
+                if (!IsWhitespace(inputBuffer[i]))
+                {
+                    tmpBuffer[count++] = inputBuffer[i];
+                }
             }
 
-            return temp;
+            return tmpBuffer.Slice(0, count);
+        }
+
+        private static bool IsWhitespace(byte value)
+        {
+            // We assume ASCII encoded data. If there is any non-ASCII char, so it is invalid
+            // Base64 and will be caught during decoding.
+
+            // SPACE        32
+            // TAB           9
+            // LF           10
+            // VTAB         11
+            // FORM FEED    12
+            // CR           13
+
+            // https://github.com/dotnet/coreclr/issues/914
+            return value == 32 || ((uint)value - 9 <= (13 - 9)) ? true : false;
```

Comment:
Is the comment justifying the ternary?  If so, please explain that.  Especially since the title of the issue is about string.IsNullOrEmpty.

Reply:
Codegen with the ternary is better. I'll add a comment to point this out.